### PR TITLE
DEV-5218: Show 404 if magic link page fails to load

### PR DESCRIPTION
**Pull request by GuiMend**
Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)
n/a
#### What's this PR do?
- Shows page not found 404 if Portal page request for Live Page fails with 404
- Shows generic error if Portal page request for Live Page fails with any other errors
- Rework component `LivePage404` to be a more generic error component (`PageError`)
- Migrate some JS files to TS (Dashboard.tsx
- Improve TS in AvatarMenu, DashboardTopbar
#### Why are we doing this? How does it help us?
Improve user experience around New Portal
#### Are there detailed, specific, step-by-step testing instructions in the associated ticket?
yes
#### Did this PR make changes to the user interface that may require changes to the user-facing docs?
yes
#### Have automated unit tests been added? If not, why?
yes
#### How should this change be communicated to end users?
Non existent Portal pages will now show 404
#### Are there any smells or added technical debt to note?
no
#### Has this been documented? If so, where?
no
#### What are the relevant tickets? Add a link to any relevant ones.
https://news-revenue-hub.atlassian.net/browse/DEV-5218
#### Do any changes need to be made before deployment to production (adding environment variables, for example)? If so, open a ticket and link it to the ticket for this PR and list it here:
no
#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:
no

### CODE REVIEW

File: spa/src/components/common/PageError/PageError.tsx
Comment (nrh-cklimas) [inline spa/src/components/common/PageError/PageError.tsx:1]: Could this appear in Storybook?

File: spa/src/components/common/PageError/PageError.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import PropTypes from 'prop-types';
+import { Description, PageErrorWrapper, StatusCode, Wrapper } from './PageError.styled';
+
+export const FUNDJOURNALISM_404_REDIRECT = 'https://fundjournalism.org/?utm_campaign=404#donate';
+
+const PageErrorPropTypes = {
+  showRedirect: PropTypes.bool,
+  statusCode: PropTypes.number,
+  errorMessage: PropTypes.string
+};
```
Comment (nrh-cklimas) [inline spa/src/components/common/PageError/PageError.tsx:15]: How would you feel about making this a bit more generic and flexible with these props:

```
header: PropTypes.node,
description: PropTypes.node
```

And then rendering below

```tsx
<Wrapper>
  {header && <Header>header</Header>}
  <Description>{description}</Description>
</Wrapper>
```

File: spa/src/components/common/PageError/PageError.test.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import { render, screen } from 'test-utils';
+import PageError, { PageErrorProps } from './PageError';
+
+describe('PageError', () => {
```
Comment (nrh-cklimas) [inline spa/src/components/common/PageError/PageError.test.tsx:4]: Should either have a test around setting analytics config, or move that out of the component. I lean towards moving it out of the component because it may not be needed everywhere.

File: spa/src/utilities/hasContributionsSectionAccess.ts [hunk 3]
```diff
@@ -5,10 +5,10 @@ import {
 import { User } from 'hooks/useUser.types';
 import flagIsActiveForUser from './flagIsActiveForUser';
 
-function hasContributionsSectionAccess(user: User) {
+function hasContributionsSectionAccess(user?: User) {
```
Comment (nrh-cklimas) [inline spa/src/utilities/hasContributionsSectionAccess.ts:8]: Could you explain why this change is here? I don't see it referred to elsewhere in the PR.

File: spa/src/components/portal/PortalPage.tsx [hunk 4]
```diff
@@ -33,8 +35,18 @@ function PortalPage({ children, className }: PortalPageProps) {
...
+    (pageError as AxiosError)?.response?.status === 404
+      ? {
+          statusCode: 404,
+          errorMessage: "The page you requested can't be found."
+        }
+      : {
+          errorMessage: 'Something went wrong. Please try again later.'
+        };
+
+  // If rp has no default page, show error
```
Comment (nrh-cklimas) [inline spa/src/components/portal/PortalPage.tsx:48]: Is this comment accurate still? I think it's referring to a loading state.

File: spa/src/utilities/hasContributionsSectionAccess.ts (hunk 3 above)
Comment (GuiMend) [reply spa/src/utilities/hasContributionsSectionAccess.ts:8]: It's because of this usage in `Dashboard.tsx`. I transformed the dashboard from JS to TS and I had to fix the `hasContributionsSectionAccess` 
image

File: spa/src/components/common/PageError/PageError.test.tsx (hunk 2 above)
Comment (nrh-cklimas) [reply spa/src/components/common/PageError/PageError.test.tsx:4]: I think this didn't get addressed as of the latest commit on the PR.

Comment (GuiMend) [reply spa/src/components/common/PageError/PageError.test.tsx:4]: My bad Chris. I did make sure that we are setting analytics where needed, but forgot to remove it from PageError. Fixed it now.

File: spa/src/utilities/hasContributionsSectionAccess.ts (hunk 3 above)
Comment (GuiMend) [reply spa/src/utilities/hasContributionsSectionAccess.ts]: [6c58749](https://github.com/newsrevenuehub/rev-engine/pull/1641/commits/6c58749d9d0d96883dbc00eb832246cc8ea17cff)

File: spa/src/components/common/PageError/PageError.test.tsx (hunk 2 above)
Comment (GuiMend) [reply spa/src/components/common/PageError/PageError.test.tsx:4]: [6c58749](https://github.com/newsrevenuehub/rev-engine/pull/1641/commits/6c58749d9d0d96883dbc00eb832246cc8ea17cff)

**Review by nrh-cklimas — APPROVED** [review]
